### PR TITLE
Update logging to be better traceable and more uniform

### PR DESCRIPTION
**Pull request by saschagrunert**
This commit updates the logging by adding a new `log()` function, which
adds the `origin=OCICNI` fields to all logs. This enables us to hint to
this library if a log occurs in API consumers. Beside this, most logs
start now with a lowercase letter and the overall verbosity has been
decreased (info → debug).

---

Example output in CRI-O:
```
INFO[2019-07-19 09:38:44.341781159+02:00] got pod network &{Name:kube-dns-79c9fb97c8-4j9sq Namespace:kube-system ID:e9e85dce43a4d7d3adfb5875bd0376836eeea4cb3f58a3e27e5f137e004a1c24 NetNS:/proc/17091/ns/net Networks:[] RuntimeConfig:map[crio:{IP: PortMappings:[] Bandwidth:<nil> IpRanges:[]}]}  origin=OCICNI
```

### CODE REVIEW

File: pkg/ocicni/ocicni.go [hunk 1]
```diff
@@ -261,34 +261,34 @@ func loadNetworks(exec cniinvoke.Exec, confDir string, binDirs []string) (map[st
...
-			logrus.Warningf("CNI config list %s has no networks, skipping", confFile)
+			log().Warnf("CNI config list %s has no networks, skipping", confFile)
 			continue
 		}
 		if confList.Name == "" {
 			confList.Name = path.Base(confFile)
 		}
 
-		logrus.Infof("Found CNI network %s (type=%v) at %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
+		log().Debugf("found CNI network %s (type=%v) at %s", confList.Name, confList.Plugins[0].Network.Type, confFile)
```
Comment (dcbw) [inline pkg/ocicni/ocicni.go]: Any reason this switched from Infof() to Debugf() in this PR? At least until one of the reload PRs lands, this shouldn't be called often and it's very useful information.

File: pkg/ocicni/ocicni.go [hunk 2]
```diff
@@ -568,22 +568,22 @@ func (network *cniNetwork) checkNetwork(cacheDir string, podNetwork *PodNetwork,
...
 	rt, err := buildCNIRuntimeConf(cacheDir, podNetwork, ifName, runtimeConfig)
 	if err != nil {
-		logrus.Errorf("Error deleting network: %v", err)
+		log().Errorf("building CNI runtime config: %v", err)
 		return err
 	}
 
 	netconf, cninet := network.NetworkConfig, network.CNIConfig
-	logrus.Infof("About to del CNI network %s (type=%v)", netconf.Name, netconf.Plugins[0].Network.Type)
+	log().Debugf("deleting CNI network %s (type=%v)", netconf.Name, netconf.Plugins[0].Network.Type)
```
Comment (dcbw) [inline pkg/ocicni/ocicni.go]: Same here; I think this PR should just do the conversion to log() and then we can have a discussion about log levels and verbosity somewhere else.

**Comment by dcbw** [comment]
@saschagrunert I'd merge this PR today except for the change of log levels for some functions... if we really do want to change the verbosity I think that's a separate discussion.

File: pkg/ocicni/ocicni.go (hunk 1 above)
Comment (saschagrunert) [reply pkg/ocicni/ocicni.go]: Okay, I will revert it back to `Infof`

File: pkg/ocicni/ocicni.go (hunk 2 above)
Comment (saschagrunert) [reply pkg/ocicni/ocicni.go]: Yep, I will revert it back to `Infof`

**Comment by saschagrunert** [comment]
> @saschagrunert I'd merge this PR today except for the change of log levels for some functions... if we really do want to change the verbosity I think that's a separate discussion.

I agree and changed it as suggested.

File: pkg/ocicni/ocicni.go [hunk 3]
```diff
@@ -476,15 +476,15 @@ func (plugin *cniNetworkPlugin) GetPodNetworkStatus(podNetwork PodNetwork) ([]cn
...
 	rt, err := buildCNIRuntimeConf(cacheDir, podNetwork, ifName, runtimeConfig)
 	if err != nil {
-		logrus.Errorf("Error adding network: %v", err)
+		log().Errorf("adding network: %v", err)
 		return nil, err
 	}
 
 	netconf, cninet := network.NetworkConfig, network.CNIConfig
-	logrus.Infof("About to add CNI network %s (type=%v)", netconf.Name, netconf.Plugins[0].Network.Type)
+	log().Debugf("adding CNI network %s (type=%v)", netconf.Name, netconf.Plugins[0].Network.Type)
```
Comment (dcbw) [inline pkg/ocicni/ocicni.go]: @saschagrunert can we change this one back to Infof() as well?

Comment (saschagrunert) [reply pkg/ocicni/ocicni.go]: Yes, I also rebased on top of the latest master to resolve the conflicts.

**Comment by saschagrunert** [comment]
@dcbw @haircommander @mrunalp  may I ask you to give it another look? :)

**Comment by haircommander** [comment]
LGTM

**Comment by rhatdan** [comment]
@saschagrunert What is the scoop with this PR?

**Comment by saschagrunert** [comment]
Let's close this one for now.

**Pull request closed without merging**